beta/src/components/MDX/Challenges/Challenges.tsx

```diff
@@ -79,6 +80,26 @@ export function Challenges({
   const queuedScrollRef = useRef<boolean>(false);
   const [activeIndex, setActiveIndex] = useState(0);
   const currentChallenge = challenges[activeIndex];
+  const {asPath} = useRouter();
+  const [isMounted, setMounted] = useState(false);
+
+  useEffect(() => {
+    const challengeIndex = challenges.findIndex(
+      (challenge) => challenge.id === asPath.split('#')[1]
+    );
+    if (challengeIndex !== -1) {
+      if (isMounted) {
+        scrollAnchorRef.current!.scrollIntoView({
+          block: 'start',
+        });
+      } else {
+        setActiveIndex(challengeIndex);
+      }
+    }
+    if (!isMounted) {
+      setMounted(true);
```

Comment:
This seems unnecessary since `isMounted` is never used for rendering. We should avoid triggering a re-render since there's no actual visual change. Let's keep this boolean in a ref instead.

Reply:
Updated! That's totally my bad, I remember the recommendation of the ref usage in beta docs but failed to react as the first option to use.